contracts/src/BurnCollector.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+interface IHypNativeMinter {
```

Comment:
We can also consider moving this impl to this repo or somewhere else as we spoke about.

I'm happy to do that in a following PR tho if we decide that. Let's wait and see